### PR TITLE
Implements GameWindowTransitionsHandler layout.

### PR DESCRIPTION
**Pull request by tomsons26**

### CODE REVIEW

File: src/game/client/gui/gamewindowtransitions.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    m_group3 = nullptr;
+    m_group4 = nullptr;
+}
+
+void GameWindowTransitionsHandler::Reset()
+{
+    m_group1 = nullptr;
+    m_group2 = nullptr;
+    m_group3 = nullptr;
+    m_group4 = nullptr;
```
Comment (xezon) [inline src/game/client/gui/gamewindowtransitions.cpp:32]: What about `m_groupList` on Reset?

File: src/game/client/gui/gamewindowtransitions.h [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    TRANSITION_FULL_FADE,
+    TRANSITION_TEXT_ON_FRAME,
+    TRANSITION_MAIN_MENU_MEDIUM_SCALE_UP,
+    TRANSITION_MAIN_MENU_SMALL_SCALE_DOWN,
+    TRANSITION_CONTROL_BAR_ARROW,
+    TRANSITION_SCORE_SCALE_UP,
+    TRANSITION_REVERSE_SOUND,
+};
+
+class TransitionGroup;
```
Comment (xezon) [inline src/game/client/gui/gamewindowtransitions.h:40]: You can move this up to the other class declarations.

File: src/game/client/gui/gamewindowtransitions.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ *            LICENSE
+ */
+#pragma once
+#include "always.h"
+#include "namekeygenerator.h"
+#include "subsysteminterface.h"
+#include <list>
+
+class GameWindow;
+class Transition;
```
Comment (xezon) [inline src/game/client/gui/gamewindowtransitions.h]: `GameWindow` and `Transition` classes are not used in this file (yet). Can be removed?

File: src/game/client/gui/gamewindowtransitions.h (hunk 2 above)
Comment (tomsons26) [reply src/game/client/gui/gamewindowtransitions.h:40]: the actual class will replace that later

File: src/game/client/gui/gamewindowtransitions.cpp (hunk 1 above)
Comment (tomsons26) [reply src/game/client/gui/gamewindowtransitions.cpp:32]: it doesn't do anything with it but added a todo to invesigate

**Review by xezon — APPROVED** [review]

**Pull request merged**